src/main/java/ch/njol/skript/lang/SkriptEvent.java

```diff
@@ -52,6 +55,20 @@ public abstract class SkriptEvent extends Structure {
 
 	public static final Priority PRIORITY = new Priority(600);
 
+	/**
+	 * Checks if the current event is of the given types. Will log an error if false.
+	 * @param events the events to check.
+	 * @return true if the current event is of the given types.
+	 */
+	@SafeVarargs
+	public static boolean isEvent(Class<? extends Event>... events) {
+		ParserInstance parser = ParserInstance.get();
+		if (CollectionUtils.containsAny(parser.getCurrentEvents(), events))
```

Comment:
I would use this method to ensure the check is being performed properly https://github.com/SkriptLang/Skript/blob/e8b6d3900a17c0c4be10c17ff8b037aeece5df9c/src/main/java/ch/njol/skript/lang/parser/ParserInstance.java#L250

Reply:
What do you mean by this sorry?

Reply:
I think pickle meant like this `if (parser.isCurrentEvent(events))`

Reply:
Ahhhh i see, sorry shall do this now